packages/react-bindings/src/useStatsigInternalClientFactoryAsync.ts

```diff
@@ -14,24 +14,25 @@ export function useStatsigInternalClientFactoryAsync<T extends StatsigClient>(
   args: FactoryArgs,
 ): { isLoading: boolean; client: T } {
   const [isLoading, setIsLoading] = useState(true);
-  const clientRef = useRef<T | null>(_getInstance(args.sdkKey) as T | null);
-
-  const client = useMemo(() => {
-    if (clientRef.current) {
-      return clientRef.current;
+  const clientRef = useRef<T>(
+    (_getInstance(args.sdkKey) as T | null) ?? factory(args),
+  );
+
+  useEffect(() => {
+    // already initializing or initialized
+    if (clientRef.current.loadingStatus !== 'Uninitialized') {
```

Comment:
Can you better outline what it is you are trying to solve? This change means that in strict mode, we will immediately render, regardless of the SDKs status.

Adding this logic means that we will just set isLoading to false on the re-render as its loadingStatus will be "Loading" and therefore pass this check.

Reply:
This logic is intended to prevent multiple SDK clients/network requests being created.

In React Strict mode, a re-render occurs, blowing away the previous state.
The logic here is supposed to create the client on the first render, then reuse it on the second.

Reply:
Correct. 
As far as I could tell, the old logic tried to do the same.

Reply:
This is incorrect. The loadingStatus goes from "Uninitialized" -> "Loading" -> "Ready". Do this here means that we would potentially `setIsLoading(false)` when loading status is "Loading"